pkg/cmd/cli/config/store.go

```diff
@@ -116,13 +116,15 @@ func (c *ConfigStore) SaveToFile(credentialsName string, namespace string, clien
 
 	credentials := clientcmdapi.NewAuthInfo()
 	credentials.Token = clientCfg.BearerToken
-	credentials.ClientCertificate = clientCfg.TLSClientConfig.CertFile
-	if len(credentials.ClientCertificate) == 0 {
-		credentials.ClientCertificateData = clientCfg.TLSClientConfig.CertData
-	}
-	credentials.ClientKey = clientCfg.TLSClientConfig.KeyFile
-	if len(credentials.ClientKey) == 0 {
-		credentials.ClientKeyData = clientCfg.TLSClientConfig.KeyData
+	if len(credentials.Token) == 0 {
```

Comment:
Should that be something that login (with or without a flag) should avoid?

----- Original Message -----
> > @@ -116,13 +116,15 @@ func (c *ConfigStore) SaveToFile(credentialsName
> > string, namespace string, clien
> >  
> >  	credentials := clientcmdapi.NewAuthInfo()
> >  	credentials.Token = clientCfg.BearerToken
> > -	credentials.ClientCertificate = clientCfg.TLSClientConfig.CertFile
> > -	if len(credentials.ClientCertificate) == 0 {
> > -		credentials.ClientCertificateData = clientCfg.TLSClientConfig.CertData
> > -	}
> > -	credentials.ClientKey = clientCfg.TLSClientConfig.KeyFile
> > -	if len(credentials.ClientKey) == 0 {
> > -		credentials.ClientKeyData = clientCfg.TLSClientConfig.KeyData
> > +	if len(credentials.Token) == 0 {
> 
> Just to make it clear, we are not clearing certs, and we are creating a new
> user in the config already. The problem is that the cert from the previous
> user in config is being carried to the new user and saved along with the
> token.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1343/files#r27515444